std/machines/large_field/rotate.asm

```diff
@@ -0,0 +1,73 @@
+use std::utils::unchanged_until;
+use std::utils::cross_product;
+use std::convert::int;
+use std::check::require_field_bits;
+
+/// Rotate on single byte input using an exhaustive table, returning 32-bit value as result.
+/// TODO this way, we cannot prove anything that rotates by more than 31 bits.
```

Comment:
I would just turn this into a comment that says that we can rotate by at most 31.